lib/wca_live/scoretaking.ex

```diff
@@ -224,41 +224,61 @@ defmodule WcaLive.Scoretaking do
     if Round.open?(round) do
       {:error, "cannot open this round as it is already open"}
     else
-      Multi.new()
-      |> Multi.run(:finish_previous, fn _repo, _changes ->
-        if round.number > 1 do
-          previous = get_previous_round(round) |> Repo.preload(:results)
-          finish_round(previous)
-        else
-          {:ok, nil}
+      message = validate_previous_rounds(round)
+      if message do
```

Comment:
To avoid nesting you can do this:

```elixir
cond do
  Round.open?(round) ->
    {:error, ...}

  message = validate_previous_rounds(round) ->
    {:error, error}

  true ->
    Multi.new()
    |> ...
end
```